Vision/SplitImage/SplitImage.cs

```diff
@@ -47,16 +55,36 @@ static SplitImage()
 
             WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
                 (inRecord, outRecord) => {
-                    var imageUrl = inRecord.Data["imageLocation"] as string; // no url parameters
-                    var sasToken = inRecord.Data["sasToken"] as string; // includes ? to start query params
+                    object imageUrlObject = null;
+                    object sasTokenObject = null;
+                    inRecord.Data.TryGetValue("imageLocation", out imageUrlObject);
+                    inRecord.Data.TryGetValue("sasToken", out sasTokenObject);
 
-                    string fullUri = imageUrl + sasToken;
-                    JArray splitImages = new JArray();
+                    string imageUrl = imageUrlObject as string;
+                    string sasToken = sasTokenObject as string;
 
+                    if (string.IsNullOrWhiteSpace(imageUrl))
+                    {
+                        outRecord.Errors.Add(new WebApiErrorWarningContract() { Message = "Parameter 'imageUrl' is required to be present and a valid uri." });
+                        return outRecord;
+                    }
+
+                    JArray splitImages = new JArray();
 
                     using (WebClient client = new WebClient())
                     {
-                        var fileData = client.DownloadData(new Uri(fullUri));
+                        byte[] fileData = new byte[0];
```

Comment:
nit: `Array.Empty<byte>()` is less allocate-y, but I'd forgo the initial value assignment altogether since it's going to immediately be thrown away, and use a ternary instead.